bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java

```diff
@@ -982,76 +1009,70 @@ public void testEnforceMinNumFaultDomainsForWrite() throws Exception {
-
-        // Assign all initial bookies started to the default rack by modifying the `localhost` in the StaticDNSResolver
-        StaticDNSResolver.reset();
-        StaticDNSResolver.addNodeToRack("localhost", NetworkTopology.DEFAULT_REGION + "/rack0");
-
-        try (BookKeeperTestClient bk = new BookKeeperTestClient(conf)) {
-            // Modify localhost in StaticDNSResolver to assign next started bookie to a different rack("/rack1")
-            StaticDNSResolver.addNodeToRack("localhost", NetworkTopology.DEFAULT_REGION + "/rack1");
-            List<BookieSocketAddress> bookieRack1 = Collections.singletonList(startNewBookieAndReturnAddress());
+        MockRackawareEnsemblePlacementPolicy currPlacementPolicy =
+                (MockRackawareEnsemblePlacementPolicy) bk.getPlacementPolicy();
+        currPlacementPolicy.resetCounter();
+        BookieSocketAddress bookieToSleep;
 
-            try (LedgerHandle lh = bk.createLedger(3, 3, digestType, password)) {
-                CountDownLatch sleepLatch = new CountDownLatch(1);
+        try (LedgerHandle lh = bk.createLedger(3, 3, 2, digestType, password)) {
+            CountDownLatch sleepLatch = new CountDownLatch(1);
 
-                Thread bookieSleeperCountdown = new Thread(() -> {
-                    try {
-                        LOG.info("Counting down 10 seconds before awakening non default rack bookie");
-                        sleepLatch.await(10, TimeUnit.SECONDS);
-                        LOG.info("Non default rack bookie awake");
-                    } catch (InterruptedException ignored) {}
+            Thread bookieSleeperCountdown = new Thread(() -> {
+                try {
+                    LOG.info("Counting down 10 seconds before waking sleeping bookie");
+                    sleepLatch.await(10, TimeUnit.SECONDS);
+                    LOG.info("Picked bookie awake");
+                } catch (InterruptedException ignored) {
+                }
 
-                    sleepLatch.countDown();
-                });
+                sleepLatch.countDown();
+            });
 
-                Thread writeToLedger = new Thread(() -> {
-                    try {
-                        LOG.info("Initiating write for entry");
-                        long entryId = lh.addEntry(data);
-                        LOG.info("Wrote entry with entryId = {}", entryId);
-                        assertTrue(entryId >= 0);
-                    } catch (InterruptedException | BKException ignored) {
-                        // Fail the test if the write times out
-                        fail("Write should not have thrown error");
-                    }
-                });
+            Thread writeToLedger = new Thread(() -> {
+                try {
+                    LOG.info("Initiating write for entry");
+                    long entryId = lh.addEntry(data);
+                    LOG.info("Wrote entry with entryId = {}", entryId);
+                    assertTrue(entryId >= 0);
```

Comment:
junit asserts work only in main thread..so these threads are non-functional

Reply:
also why comparing >= 0? it should be just == 0, right?

Reply:
Modifying.